src/core/integrator_wrappers.jl

```diff
@@ -0,0 +1,195 @@
+state `u` by using `reinit!(smap, u)` and then calling `step!` as normally.
+
+## Keyword Arguments
+* `u0`: initial state
+* `diffeq` is a `NamedTuple` (or `Dict`) of keyword arguments propagated into
+  `init` of DifferentialEquations.jl.
+
+
+## Example
+```julia
+F = 0.27; ω = 0.1;  # smooth boundary
+ds = Systems.duffing(zeros(2), ω = ω, f = F, d = 0.15, β = -1)
+smap = stroboscopicmap(ds, 2*pi/ω; diffeq = (;reltol = 1e-8))
+reinit!(smap,[1., 1.])
+step!(smap)
+u = get_state(smap)
+```
+"""
+function stroboscopicmap(ds::CDS{IIP, S, D}, T = nothing; u0 = get_state(ds),
+	diffeq = NamedTuple(), kwargs...
+	) where {IIP, S, D}
+
+    if !isempty(kwargs)
+        @warn DIFFEQ_DEP_WARN
+        diffeq = NamedTuple(kwargs)
+    end
+
+	if isnothing(T)
+		@warn "T must be defined, taking T=1 as default"
+		T =1
+	end
+
+	integ = integrator(ds, u0; diffeq)
+	return StroboscopicMap(integ, T)
+end
+
+mutable struct StroboscopicMap{I, F}
+	integ::I
+	T::F
+end
+
+function DynamicalSystemsBase.step!(smap::StroboscopicMap)
+	step!(smap.integ, smap.T, true)
+	return smap.integ.u
+end
+function DynamicalSystemsBase.reinit!(smap::StroboscopicMap, u0)
+	reinit!(smap.integ, u0)
+	return
+end
+function DynamicalSystemsBase.get_state(smap::StroboscopicMap)
+	return smap.integ.u
+end
+
+function Base.show(io::IO, smap::StroboscopicMap)
+    println(io, "Iterator of the Stroboscopic map")
+    println(io,  rpad(" rule f: ", 14),     DynamicalSystemsBase.eomstring(smap.integ.f.f))
+    println(io,  rpad(" Period: ", 14),     smap.T)
+end
+
+
+
+"""
+	projectedsystem(ds::ContinuousDynamicalSystem, Δt; kwargs...)  → psys
+
+Returns an integrator that produces iterations of the dynamical system `ds` on a
+projected subspace.
+
+You can progress the map one step on the section by calling `step!(psys)`,
+which also returns the next state. You can also set the integrator to start from a new
+state `u` by using `reinit!(psys, u)` and then calling `step!` as normally.
+
+## Keyword Arguments
+* `u0`: initial state
+* `idxs = 1:length(Dp)`: This vector selects the variables of the system that will define the
+  subspace the dynamics will be projected into.
+* `complete_state = zeros(D-Dp)`: This argument allows setting the _remaining_ variables
+  of the dynamical system state on each initial condition `u`, with `Dp` the dimension
+  of the projected subspace. It can be either a vector of length `D-Dp`, or a function `f(y)` that
+  returns a vector of length `D-Dp` given the _projected_ initial condition on the grid `y`.
+* `diffeq` is a `NamedTuple` (or `Dict`) of keyword arguments propagated into
+  `init` of DifferentialEquations.jl.
+
+
+## Example
+```julia
+ds = Systems.lorenz_iip()
+psys = projectedsystem(ds, 0.1; idxs = 1:2, complete_state=[0.0], diffeq = (;reltol = 1e-8))
+reinit!(psys,[1., 1.])
+step!(psys)
+u = get_state(psys)
+```
+"""
+function projectedsystem(ds::CDS{IIP, S, D}, Δt = nothing;  u0 = get_state(ds),
```

Comment:
I think it may be better to create a struct `ProjectedSystem` that contains the complete state business. Then define on it the `integrator` method that takes as an input `Δt, diffeq` and creates a `ProjectedIntegratror` which is the current struct. Seems cleaner and makes a bit more sense as well given the name (a projected system doesn't convey that it is an integrator).